src/components/Contentful/DynamicHeroClassic.vue

```diff
@@ -0,0 +1,335 @@
+									:alt="responsiveHeroDescription"
+									:source-sizes="responsiveHeroImages"
+								/>
+							</router-link>
+						</template>
+						<template v-if="isHeroVideo">
+							<video
+								:src="heroMedia[0].url"
+								class="contentful-hero__video-wrapper--video"
+								autoplay
+								loop
+								muted
+								playsinline
+							></video>
+						</template>
+					</div>
+
+					<div
+						class="tw-col-span-12 md:tw-col-span-6 xl:tw-col-span-7"
+						:class="{ 'tw-order-first': swapOrder }"
+					>
+						<h1
+							v-if="heroHeadline"
+							v-html="heroHeadline"
+							class="tw-mb-2 md:tw-mb-3"
+						></h1>
+						<h2
+							v-if="heroSubHeadline"
+							v-html="heroSubHeadline"
+							class="tw-mb-2 md:tw-mb-3"
+						></h2>
+						<div v-if="heroBody" class="tw-prose tw-mb-2 md:tw-mb-3">
+							<dynamic-rich-text :html="heroBody" />
+						</div>
+						<kv-button
+							v-if="buttonText"
+							class="tw-w-full md:tw-w-auto"
+							:class="`${buttonClass}`"
+							:to="buttonTo"
+							:variant="buttonStyle"
+							@click.native="buttonClick"
+							v-kv-track-event="buttonAnalytics"
+						>
+							{{ buttonText }}
+						</kv-button>
+					</div>
+				</kv-grid>
+			</kv-page-container>
+		</template>
+	</section-with-background-classic>
+</template>
+
+<script>
+import SectionWithBackgroundClassic from '@/components/Contentful/SectionWithBackgroundClassic';
+import { richTextRenderer } from '@/util/contentful/richTextRenderer';
+import DynamicRichText from '@/components/Contentful/DynamicRichText';
+import KvButton from '~/@kiva/kv-components/vue/KvButton';
+import KvContentfulImg from '~/@kiva/kv-components/vue/KvContentfulImg';
+import KvGrid from '~/@kiva/kv-components/vue/KvGrid';
+import KvPageContainer from '~/@kiva/kv-components/vue/KvPageContainer';
+
+/**
+* Dynamic Hero Component
+* This component will display a Hero driven by a Contentful Content Group
+* Depending on the number and type of media attached to the content group
+* on contentful, the left side will either be an image, a video or a carousel.
+* */
+
+export default {
+	components: {
+		KvButton,
+		KvCarousel: () => import('@/components/Kv/KvCarousel'),
+		KvCarouselSlide: () => import('@/components/Kv/KvCarouselSlide'),
+		DynamicRichText,
+		KvContentfulImg,
+		KvGrid,
+		KvPageContainer,
+		SectionWithBackgroundClassic,
+	},
+	props: {
+		/**
+		 * Content group content from Contentful
+		* */
+		content: {
+			type: Object,
+			default: () => {},
+		},
+	},
+	computed: {
+		background() {
+			return this.content?.contents?.find(({ contentType }) => {
+				return contentType ? contentType === 'background' : false;
+			});
+		},
+		buttonAnalytics() {
+			const defaults = ['Hero', 'click-hero-cta', this.buttonText];
+			const contentfulAnaltyicsEvent = this.buttonContent?.analyticsClickEvent ?? null;
+			return contentfulAnaltyicsEvent || defaults;
+		},
+		// TODO: Consider Deprecating
+		buttonClass() {
```

Comment:
This was used to listen to the button for custom actions, like the MG interactive selector, not sure we can deprecate easily

Reply:
Oh good catch that's what we needed to know!